[FEAT] Allow specify global proxy with environment variable for backend

TTsdzb opened the pull request:
Allow the user to specify a proxy server for backend requests (the ones using axios) with normal proxy envs like `http_proxy`, `https_proxy` or `all_proxy`.

This allows the user to specify a global proxy server for making requests. Provides a solution for #3085. This is essential for some poorly routed or DNS polluted area (like Mainland China), where users may have trouble connecting to Github or game stores, or have really poor download speed. In this case they have to use a proxy server to get them work.

This may not be the best solution but should be the simplest one.

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

TTsdzb committed:
[FEAT] Allow specify global proxy with environment variable
Allow the user to specify a proxy server for backend requests with `GLOBAL_AGENT_HTTP_PROXY`.

flavioislima commented:
All contributors have signed the CLA  ✍️ ✅<br/><sub>Posted by the ****CLA Assistant Lite bot****.</sub>

TTsdzb commented:
I have read the CLA Document and I hereby sign the CLA

TTsdzb committed:
[FEAT] Use standard proxy envs
Use standard proxy environment variables when given, instead of
customized ones by the lib.

arielj commented:
> However when I tried to login, it fails with following error. I think I'm missing some relevant toolsets, and since this also happens before I make these changes, I assume it's not caused by my changes. Please let me know if I'm missing something or you get into any problem.
> 
> ```
> Error running command "LEGENDARY_CONFIG_PATH=/home/liusq/.config/heroic/legendaryConfig/legendary /tmp/.mount_Heroic3PInp1/resources/app.asar.unpacked/build/bin/x64/linux/legendary auth --code <redacted>": Error: spawn ./legendary ENOENT
> at Process.ChildProcess._handle.onexit (node:internal/child_process:286:19)
> at onErrorNT (node:internal/child_process:484:16)
> at processTicksAndRejections (node:internal/process/task_queues:82:21)
> ```

looks like you are missing the binaries? are you generating an app image? make sure you run `pnpm download-helper-binaries` first, and you don't really need to generate an app image to test things locally though if you are doing that

TTsdzb commented:
> make sure you run `pnpm download-helper-binaries` first

This make everything works, thanks. It works smoothly for me and I think it's ready for review now.

TTsdzb commented:
Btw, debugging with `pnpm start` needs these binaries to be downloaded, too. Could you guys please mention it in the README.md?
